src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,88 @@
+package duke.storage;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+import duke.dukeexceptions.DukeExceptions;
+import duke.tasklist.TaskList;
+import duke.tasks.Task;
+
+/**
+ * Class to abstract the interaction between local storage and program task list.
+ */
+public class Storage {
+    private File file;
+    private final String path;
+
+    /**
+     * Constructor for storage abstraction.
+     *
+     * @param path the data path where we will load and store the data
+     */
+    public Storage(String path) {
+        this.path = path;
+        fileSetup();
+    }
+
+    /**
+     * Sets up file and directory if not present.
+     */
+    public void fileSetup() {
+        File directory = new File(path);
+        try {
+            if (!directory.exists()) {
+                directory.mkdir();
+            }
+
+            this.file = new File(path, "duke.txt");
+
+            if (!this.file.exists()) {
+                this.file.createNewFile();
+            }
+
+
+        } catch (IOException exception) {
+            System.out.println(exception.getMessage());
```

Comment:
So what happens when an exception is encountered during the try block? Where is it thrown to, and can the program be recovered?